common/src/verify.rs

```diff
@@ -21,34 +24,39 @@ pub enum Error {
 pub fn verify_header_to_header(h1: &BlockHeader, h2: &BlockHeader) -> Result<(), Error> {
     if h2.height != h1.height + 1 {
         return Err(Error::InvalidArgument(format!(
-            "Invalid height: expected {}, got {}",
+            "invalid height: expected {}, got {}",
             h1.height + 1,
             h2.height
         )));
     }
     if h2.previous_hash != h1.to_hash256() {
         return Err(Error::InvalidArgument(format!(
-            "Invalid previous hash: expected {}, got {}",
+            "invalid previous hash: expected {}, got {}",
             h1.to_hash256(),
             h2.previous_hash
         )));
     }
-    if !h2.validator_set.iter().any(|(pk, _)| pk == &h1.author) {
-        return Err(Error::InvalidArgument(format!(
-            "Invalid author: got {}",
-            h2.author
-        )));
-    }
+    verify_validator(&h1.validator_set, &h2.author)?;
     if h2.timestamp <= h1.timestamp {
         return Err(Error::InvalidArgument(format!(
-            "Invalid timestamp: expected larger than {}, got {}",
+            "invalid timestamp: expected larger than {}, got {}",
             h1.timestamp, h2.timestamp
         )));
     }
-    for signature in &h2.prev_block_finalization_proof {
-        signature.verify(h1).map_err(|e| {
-            Error::CryptoError("Invalid prev_block_finalization_proof".to_string(), e)
-        })?;
+    verify_finalization_proof(h1, &h2.prev_block_finalization_proof)?;
+    Ok(())
+}
+
+/// Verifies whether the given participant is a validator.
+pub fn verify_validator(
```

Comment:
Then how else do we verify other than inclusion check?

Reply:
Ok. I realized that checking inclusion is all about the verification.
But it seems we need this for only one time anyway. Let's do it directly instead of having a separate function.

Reply:
fixed.